src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -34,96 +33,62 @@ public EditCommand parse(String args) throws ParseException {
-        }
-        if (tutorArgMultimap.getValue(PREFIX_GENDER).isPresent()) {
-            editTutorDescriptor.setGender(ParserUtil.parseGender(tutorArgMultimap.getValue(PREFIX_GENDER).get()));
-        }
-        if (tutorArgMultimap.getValue(PREFIX_QUALIFICATION).isPresent()) {
-            editTutorDescriptor.setQualification(ParserUtil.parseQualification(
-                    tutorArgMultimap.getValue(PREFIX_QUALIFICATION).get()));
-        }
-        parseTagsForEdit(tutorArgMultimap.getAllValues(PREFIX_TAG)).ifPresent(editTutorDescriptor::setTags);
-
-        if (!editTutorDescriptor.isAnyFieldEdited()) {
-            throw new ParseException(EditCommand.MESSAGE_NOT_EDITED);
-        }
-
-        return new EditCommand(tutorIndex, editTutorDescriptor, PersonType.TUTOR);
-    }
-
-    /**
-     * Parses an edit student command into an EditCommand.
-     *
-     * @param args User input.
-     * @return An EditCommand that edits a student.
-     * @throws ParseException An exception that occurs when parsing the command to edit a student.
-     */
-    private EditCommand parseStudent(String args) throws ParseException {
-        ArgumentMultimap studentArgMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_GENDER, PREFIX_TAG);
-
-        Index studentIndex;
-
-        try {
-            studentIndex = ParserUtil.parseIndex(studentArgMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            editPersonDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
         }
-
-        EditStudentDescriptor editStudentDescriptor = new EditStudentDescriptor();
-        if (studentArgMultimap.getValue(PREFIX_NAME).isPresent()) {
-            editStudentDescriptor.setName(ParserUtil.parseName(studentArgMultimap.getValue(PREFIX_NAME).get()));
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            editPersonDescriptor.setPhone(ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get()));
         }
-        if (studentArgMultimap.getValue(PREFIX_PHONE).isPresent()) {
-            editStudentDescriptor.setPhone(ParserUtil.parsePhone(studentArgMultimap.getValue(PREFIX_PHONE).get()));
+        if (argMultimap.getValue(PREFIX_GENDER).isPresent()) {
+            editPersonDescriptor.setGender(ParserUtil.parseGender(argMultimap.getValue(PREFIX_GENDER).get()));
         }
-        if (studentArgMultimap.getValue(PREFIX_GENDER).isPresent()) {
-            editStudentDescriptor.setGender(ParserUtil.parseGender(
-                    studentArgMultimap.getValue(PREFIX_GENDER).get()));
+        if (argMultimap.getValue(PREFIX_QUALIFICATION).isPresent() && personType.equals(PersonType.TUTOR)) {
```

Comment:
Slight optimization, can put `personType.equals(PersonType.TUTOR)` first to short-circuit the evaluation.